geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/misc/WANSSLDUnitTest.java

```diff
@@ -14,107 +14,624 @@
+        }
+      }
+      fact.setDataPolicy(DataPolicy.REPLICATE);
+      fact.setScope(Scope.DISTRIBUTED_ACK);
+      fact.setOffHeap(false);
+      Region r = cache.createRegionFactory(fact.create()).create(replicatedRegionName);
+      assertNotNull(r);
+    } finally {
+      exp.remove();
+      exp1.remove();
+      exp2.remove();
+    }
+  }
 
-    vm4.invoke(() -> WANTestBase.createSender("ln", 2, false, 100, 10, false, false, null, true));
+  private void createCacheInVMs(Integer locatorPort, VM... vms) {
+    for (VM vm : vms) {
+      vm.invoke(() -> createCache(locatorPort));
+    }
+  }
 
-    vm2.invoke(
-        () -> WANTestBase.createReplicatedRegion(getTestMethodName() + "_RR", null, isOffHeap()));
+  private static void startSender(String senderId) {
+      cache.getGatewaySenders().stream()
+          .filter(s -> s.getId().equals(senderId))
+          .findFirst()
+          .ifPresent(GatewaySender::start);
+  }
 
-    vm4.invoke(() -> WANTestBase.startSender("ln"));
+  private static GatewaySenderFactory configureGateway(DiskStoreFactory dsf, File[] dirs1,
+      String dsName, int numDispatchers) {
+
+    InternalGatewaySenderFactory gateway =
+        (InternalGatewaySenderFactory) cache.createGatewaySenderFactory();
+    gateway.setParallel(false);
+    gateway.setMaximumQueueMemory(100);
+    gateway.setBatchSize(10);
+    gateway.setBatchConflationEnabled(false);
+    gateway.setManualStart(true);
+    gateway.setDispatcherThreads(numDispatchers);
+    gateway.setOrderPolicy(GatewaySender.DEFAULT_ORDER_POLICY);
+    gateway.setLocatorDiscoveryCallback(new MyLocatorCallback());
+    DiskStore store = dsf.setDiskDirs(dirs1).create(dsName);
+    gateway.setDiskStoreName(store.getName());
+    return gateway;
+  }
 
-    vm4.invoke(
-        () -> WANTestBase.createReplicatedRegion(getTestMethodName() + "_RR", "ln", isOffHeap()));
+  private void createSender(String dsName) {
+    final IgnoredException exln = IgnoredException.addIgnoredException("Could not connect");
+    try {
+      File persistentDirectory =
+          new File(dsName + "_disk_" + System.currentTimeMillis() + "_" + VM.getCurrentVMNum());
+      persistentDirectory.mkdir();
+      DiskStoreFactory dsf = cache.createDiskStoreFactory();
+      File[] dirs1 = new File[] {persistentDirectory};
+      GatewaySenderFactory gateway = configureGateway(dsf, dirs1, dsName,
+          dispatcherThreadCount);
+      gateway.create(dsName, 2);
+
+    } finally {
+      exln.remove();
+    }
+  }
 
-    vm4.invoke(() -> WANTestBase.doPuts(getTestMethodName() + "_RR", 1000));
+  private int createReceiver() {
+    GatewayReceiverFactory fact = cache.createGatewayReceiverFactory();
+    int port = AvailablePortHelper.getRandomAvailablePortForDUnitSite();
+    fact.setStartPort(port);
+    fact.setEndPort(port);
+    fact.setManualStart(true);
+    GatewayReceiver receiver = fact.create();
+    try {
+      receiver.start();
+    } catch (IOException e) {
```

Comment:
Instead of catch unexpected exception, please change the method to "throws IOException" or rethrow the IOException as an UncheckedIOException:
```
try {
  ...
} catch (IOException e) {
  throw new UncheckedIOException(e);
}
```